### DISCUSSION TITLE
error: invalid flag: -XepAllErrorsAsWarnings

### DISCUSSION THREAD

**Discussion by vivian196167** [tbroyer/gradle-errorprone-plugin#89] [Q&A] [answered]
Hi,

I am using gradle 6.8.1 and java 11. I init a gradle app project, and try to use this gradle-errorprone-plugin. it works fine until I want to make all errors showing and dot break the compile. My configuration is as bellow:

**Comment by vivian196167** [comment]
plugins {
    // Apply the application plugin to add support for building a CLI application in Java.
    id 'application'
    id("net.ltgt.errorprone") version "3.1.0"
}

repositories {
    // Use JCenter for resolving dependencies.
    jcenter()
    mavenCentral()
    maven {
      url = uri("https://plugins.gradle.org/m2/")
    }
}


dependencies {
    // Use JUnit test framework.
    // testImplementation 'junit:junit:4.13'

    // This dependency is used by the application.
    implementation 'com.google.guava:guava:29.0-jre'
    errorprone("com.google.errorprone:error_prone_core:2.12.1")
    
}

application {
    // Define the main class for the application.
    mainClass = 'come.demo.Hello.App'
}

// import net.ltgt.gradle.errorprone.errorprone
import net.ltgt.gradle.errorprone.CheckSeverity

tasks.withType(JavaCompile).configureEach {
    // options.errorprone.disableWarningsInGeneratedCode = true
    // options.errorprone.allErrorsAsWarnings = true
    options.errorprone.disableWarningsInGeneratedCode.set(true)
    options.errorprone.allErrorsAsWarnings.set(true)
    // options.compilerArgs << "-XepAllErrorsAsWarnings" 
    // options.compilerArgs.addAll(['-XDcompilePolicy=simple', '-XepAllErrorsAsWarnings'])
    options.compilerArgs += ["-XepAllErrorsAsWarnings"]
    // options.errorprone.errorproneArgs += ['-Xep:ParameterName:OFF']
    // check("CatchFail", CheckSeverity.OFF)
}

**Comment by vivian196167** [comment]
then I run as ./gradlew assemble, I got an error:
* What went wrong:
Execution failed for task ':app:compileJava'.
> error: invalid flag: -XepAllErrorsAsWarnings

Can somebody help me out?

**Comment by tbroyer** [comment, marked as answer]
`-XepAllErrorsAsWarnings` is not a compiler argument, it's an Error Prone argument (to be passed whitespace-separated as part of the `-Xplugin:ErrorProne` compiler argument); but you shouldn't have to use it when using the plugin (that's the whole idea).

Your build script should read:
```gradle
tasks.withType(JavaCompile).configureEach {
  options.errorprone {
    disableWarningsInGeneratedCode = true
    allErrorsAsWarnings = true
    // disable("ParameterName", "CatchFail")
  }
}
```
(i.e. you shouldn't even need the `CheckSeverity` enum)
